src/Features/CSharp/Portable/MakeLocalFunctionStatic/PassInCapturedVariablesAsArgumentsCodeFixProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeLocalFunctionStatic
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(PassInCapturedVariablesAsArgumentsCodeFixProvider)), Shared]
+    internal sealed class PassInCapturedVariablesAsArgumentsCodeFixProvider : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create("CS8421");
+
+        public override FixAllProvider GetFixAllProvider()
```

Comment:
:-(

This seems pretty easy to support with our SyntaxEditorBasedFixAllProvider

Reply:
Will do this in a follow up PR.

Reply:
Yes, please ensure we do this soonish while you have context on this area. Otherwise, it will likely never get picked up.

---
In reply to: [324828841](https://github.com/dotnet/roslyn/pull/37629#discussion_r324828841) [](ancestors = 324828841)

Reply:
I just made an attempt for this, but couldn't figure out how to handle multiple diagnostics in a single local function (given a single diagnostic, this codefix would fix all captures in one pass).